src/contexts/AuthContext.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useMemo, useState, useEffect, useContext, createContext } from 'react';
+
+import { getUserDetails } from 'src/sections/login/apiLogin';
+
+import { getSession, refreshTokens, isTokenExpired, clearSession } from '../utils/authUtils';
+
+const AuthContext = createContext();
+
+function AuthProvider({ children }) {
+  const backendUrl =
+    import.meta.env.VITE_BACKEND_URL || 'https://backend-service-rojjrgeqna-ue.a.run.app/';
+  const [user, setUser] = useState(null);
+  const [isAuthenticated, setIsAuthenticated] = useState(() => {
+    // Initialize isAuthenticated based on local storage to persist login state
+    const sessionData = getSession();
+    return sessionData ? !isTokenExpired(sessionData.accessTokenExpiry) : false;
+  });
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
```

Comment:
Can we use reactQuery ?

Reply:
We talked about this already in my way, which hasn't final yet. So, I am waiting for your way of using it if we are going to use it.

Reply:
Yeah,  good to go with react-query since you guys are confident about it. I've not used that much.